### ISSUE TITLE
[Translation]Add version number?

### ISSUE THREAD

**Issue by c933103** [KC3Kai/KC3Kai#794]
Would it be better for all translation file have a version number in front of them so that it can easily found out which versions had those translation files updated to? Like if the raw file is v30, and the translated file is v23, then it can easily be tell that it is needed to update from v23 to v30 instead of looking for which string do the file have and which string the file do not have,

**Comment by dragonjet** [comment]
ah this is a nice idea :+1: 
we'll probably use this when we move translations to `kc3-translations` repo

**Comment by c933103** [comment]
Translations are currently moved?

**Comment by dragonjet** [comment]
yep, though I'm planning to add this together with the new "Naval Office" system at #810, where language selection will be one whole page itself. One row per language, then has columns on version number, current translators, etc.

**Comment by dragonjet** [comment]
there might be a new single json file that would hold the versioning instead of per language directory so there's no need to call 10 ajax to 10 json files.

Also it would probably be "**last updated date**" instead of version number. it is hard to track version numbers when we can't define each version state, and how many and which of the new terms were translated.

when work starts on #810 I will check if it's possible to just check

**Comment by c933103** [comment]
well, I was talking about something that's only visible to translators so that if some language get untouched for a few weeks or so like the current tcn, after people come back they would know from where to where have each individual files have changed without doing line by line comparison and only need to read from the changes in file revision history to see which version they need to start again on and follow those file change history to translate accordingly

**Comment by dragonjet** [comment]
ah if that's the case, probably this can suffice?
https://github.com/KC3Kai/kc3-translations/commits/master/data/en/

**Comment by c933103** [comment]
I use this feature, but without version number I would need to compare each
different version to the existing one to see which version is the point
where the translation stop updating. With version numbers, it's only needed
to compare the version number
2015/09/16 11:42 "Jet Rico" <notifications@github.com>:

**Comment by dragonjet** [comment]
ah u mean the one similar to the old `README.md` file inside the translations folder where we put by date the list of terms added on that specific change?

**Comment by c933103** [comment]
Actually I get the idea from
https://github.com/KCV-Localisation/KanColleViewer-Translations , when the
main
https://github.com/KCV-Localisation/KanColleViewer-Translations/blob/master/Equipment.xml
is v.1.0.26, it is 1.0.8 in Korean translation at
https://github.com/KCV-Localisation/KanColleViewer-Translations/blob/master/ko-KR/Equipment.xml
.

With the info, you can easily findout which changes are made before last
translation and which are after by simply looking at the change in version
number.
2015/09/16 14:21 "Jet Rico" <notifications@github.com>:

**Comment by dragonjet** [comment]
This is alright if we are sure all languages will be 100% translated each time English will be updated, translators can just set the version number equal to English.

However, the reason I suggested "**last update time**" was that, we have some languages that users just send to me by file, googledoc or pastebin. Some languages even translate one file at a time. This is where I saw it will pose problems. Those languages will have their own version and is very complicated to standardize.

### Example

* `en` is for example in v1.2.34
* `de` (German) sends me `stype.json`, 1.0.1
* `pt` (Portuguese) sends me `items.json`, I also bump it to 1.0.1? 1.0.2?
* `de` now sends me a `ships.json`, I bump it to 1.0.3
* wait until `de` is complete, bump it to `1.2.34`, same as English

----
* English updates to `1.2.35` with changes to `ships.json` and `items.json`
* `de` just sends me an updates `ships.json` (but no `items.json`)
* but I cant bump `1.2.34` to `1.2.35`

----
* I could make it so that the en updates bumps the second number `1.3`, then the minor revisions of each language would be the third number, `1.3.0`
* but then English would remain `1.3.0` while those languages with more revisions end up being `1.3.9`, higher than english.

----
There will not be a standard, this is the "state" of 1.0.1, in which all should follow English, meaning the terms on `English 1.0.1` should be equal to `German 1.0.1`, etc but this will not be the case.

Versions will only help languages with active translators on GitHub, and *everything* is translated on each update, and not partially.

I may be thinking too much, but I don't want more management problems as it is :laughing:

**Comment by c933103** [comment]
yes, my original proposal is that each translation file would have their
own version. like v1.02 for stype and v.1.38 for equipment, and translators
mark where they translate up to in each file by noting the original version
number they translated till in their translated files.

But translation time might work too.
2015/09/16 15:48 "Jet Rico" <notifications@github.com>:
